### ISSUE TITLE
Cannot install pip dependencies

### ISSUE THREAD

**Issue by sebastian-luna-valero** [CERIT-SC/binderhub#3]
Hi,

When I try binder with a repository containing pip dependencies in the conda environment (e.g. https://github.com/pangeo-data/foss4g-2022/blob/main/.binder/environment.yml) I run into this issue:

```bash
PermissionError: [Errno 13] Permission denied: '/home/jovyan/.binder/condaenv.5ubl3yce.requirements.txt'
```

Do you know why?

Best regards,
Sebastian

**Comment by sebastian-luna-valero** [comment]
A bit more context:

```bash
Installing pip dependencies: ...working... failed
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<    
Traceback (most recent call last):      File "/srv/conda/lib/python3.10/site-packages/conda/exceptions.py", line 1114, 
    in __call__        return func(*args, **kwargs)     
File "/srv/conda/lib/python3.10/site-packages/conda_env/cli/main.py", line 80, 
    in do_call        exit_code = getattr(module, func_name)(args, parser)     
 File "/srv/conda/lib/python3.10/site-packages/conda_env/cli/main_update.py", line 127, 
    in execute        result[installer_type] = installer.install(prefix, specs, args, env)      
File "/srv/conda/lib/python3.10/site-packages/conda_env/installers/pip.py", line 70, 
    in install        return _pip_install_via_requirements(*args, **kwargs)      
File "/srv/conda/lib/python3.10/site-packages/conda_env/installers/pip.py", line 44, 
    in _pip_install_via_requirements        requirements = Utf8NamedTemporaryFile(mode='w',      
File "/srv/conda/lib/python3.10/site-packages/conda/auxlib/compat.py", line 54, 
    in Utf8NamedTemporaryFile        return NamedTemporaryFile(      
File "/srv/conda/lib/python3.10/tempfile.py", line 559, 
    in NamedTemporaryFile        file = _io.open(dir, mode, buffering=buffering,     
File "/srv/conda/lib/python3.10/tempfile.py", line 556, 
    in opener        fd, name = _mkstemp_inner(dir, prefix, suffix, flags, output_type)      
File "/srv/conda/lib/python3.10/tempfile.py", line 256, 
    in _mkstemp_inner        fd = _os.open(file, flags, 0o600)   
PermissionError: [Errno 13] Permission denied: '/home/jovyan/.binder/condaenv.odqxp255.requirements.txt'
```

**Comment by sebastian-luna-valero** [comment]
kaniko works when moving the dependencies out of `.binder`

xref: https://github.com/pangeo-data/foss4g-2022/pull/94

**Comment by VladimirVisnovsky** [comment]
Hi! The [issue](https://github.com/GoogleContainerTools/kaniko/issues/1524) is with Kaniko not respecting permissions along the path to files while copying in Dockerfile. I've added a small fix (3c3da997debc468b6dd6cf9b3aa8983a7063f8f0 and e61df5fb0c9d34b3a914ffcb9fd64b0f83bcfe24) where I explicitly modify permissions. Can you please confirm that it is working now?

**Comment by sebastian-luna-valero** [comment]
Hi @VladimirVisnovsky 

it works indeed, many thanks!

Sebastian
